includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -109,6 +109,8 @@ public static function normalize_url( $url ) {
 			$url = $home . $path;
 		}
 
+		$url = apply_filters( 'amp_normalize_url', $url );
```

Comment:
Let's rename `amp_normalize_url` as `amp_normalized_dimension_extractor_image_url`. That will make it more explicit (and verbose 😄).